internal/datastore/discussion_test.go

```diff
@@ -1,11 +1,987 @@
 package datastore
```

Comment:
I'm just scanning htese because they seem to be a lot of the same code. I'm sure I should read them more closely though.